tests/service/SyntaxTreeTests/SynTypeTests.fs

```diff
@@ -49,3 +49,65 @@ type T =
       ])) ->
         assertRange (3, 14) (3, 56) mTuple
     | _ -> Assert.Fail $"Could not get valid AST, got {parseResults}"
+
+[<Test>]
+let ``SynType.Or inside SynExpr.TraitCall`` () =
+    let parseResults =
+        getParseResults
+             """
+let inline (!!) (x: ^a) : ^b = ((^a or ^b): (static member op_Implicit: ^a -> ^b) x)
```

Comment:
Another case I would like to see is having 2+ "or" symbols chained:

`let inline myFunc (x: ^a) (y: ^b) : ^c = ((^a **or** ^b **or** ^c): (static member op_Implicit: ^a ->  ^c) x)`

Reply:
That makes sense, I'll add these tests.